mod_reforged/hooks/config/character.nut

```diff
@@ -25,6 +25,21 @@ local getClone = ::Const.CharacterProperties.getClone;
 ::Const.CharacterProperties.OffensiveReachIgnore <- 0;
 ::Const.CharacterProperties.BonusPerReachAdvantage <- 0;
 
+::Const.CharacterProperties.IsImmuneToStunFromPoise <- false;
```

Comment:
Probably don't need this boolean. Entities which are `IsImmuneToStun` should not get stunned by Poise. For entities which are by default immune to stun but who we want to be stunnable by Poise in our design, we can manually set their `IsImmuneToStun` to false.

Reply:
But then they get stunned by non-poise stuns like Horrified Scream, Throwing Mastery bolas, Headshots from maces with the mace perk etc. Random mods might add more skills that guarantee a stun on someone.

Reply:
I see. Yes that's a fair point. This creates a potential compatibility issue with other mods which add entities (e.g. Davkul Rising mod) which may have stun immune entities but which should be stunnable by Poise in our design. I don't think there's a way to create a compatibility layer unless we somehow utilize vanilla flags only or we create some sort of translation between them.

There are two possible compatibility layers we can make, each with its own pros and cons:
**Option 1:**
- Use the vanilla `IsImmuneToStun` flag only.
- Set that flag to `false` always.
- Entities which had that flag as true in their base properties, we give them a large PoiseMax e.g. 300 or 500 making them effectively "immune" to stun from poise.
- Advantage: Entities added by other mods are automatically assigned an appropriate Poise value and are subject to interaction with our system.
- Disadvantage: These entities are now suddenly stunnable by things which ignore poise e.g. Horrific Scream and Concussive Strikes.

**Option 2:**
- We add our own `IsImmuneToStunFromPoise` flag.
- This flag is always false. So, by default we assume all entities are stunnable from Poise.
- Entities which had `IsImmuneToStun` as true in their base properties, we give them a large PoiseMax e.g. 300 or 500 making them effectively "immune" to stun from poise.
- Advantage: Keeps the vanilla `IsImmuneToStun` flag so abilities which don't interact with the Poise system e.g. Horrific Scream, Concussive Strikes work as intended and don't need to check for a different flag.
- Disadvantage: It may cause some entities added by other mods to become "stunnable" by Poise but have a very high Poise to compensate for that.

I prefer Option 2. It is closer in its approach to what you have already implemented and I think the advantage to disadvantage weight is much better than Option 1.